sphinxcontrib/needs/needs.py

```diff
@@ -514,6 +518,21 @@ def check_configuration(_app: Sphinx, config: Config):
     extra_options = config["needs_extra_options"]
     link_types = [x["option"] for x in config["needs_extra_links"]]
 
+    external_filter = config["needs_filter_data"]
+    # Check if external filter values is really a string
+    for extern_filter, value in external_filter.items():
+        if not isinstance(value, str):
+            logger.warning(
+                "External filter value: {0} from needs_filter_data {1} is not a string.".format(value, external_filter)
+            )
+
+    # Check if needs external filter and extra option are using the same name
+    for ext_filter in external_filter:
```

Comment:
can you add .keys() here.
I know Python 3.x send back keys for dicts automatically.
But the rest of the code is always using dict.keys() to handle keys

Reply:
Done.